api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -506,6 +507,51 @@ message HttpConnectionManager {
   //
   // 3. Tracing decision (sampled, forced, etc) is set in 14th byte of the UUID.
   RequestIDExtension request_id_extension = 36;
+
+  // Configuration of local reply returned by Envoy. If Envoy needs to reject a request due to some
+  // error conditions, it calls sendLocalRely with a http status code and detail error text. If this
+  // config is not specified, the error text will be send as response body in plain text format for
+  // http, in grpc-message header for grpc. This config supports rewritting the status code and error
+  // text. The new error text can be combined from many access log operators, such as %REP(PATH)%.
+  // Its format can be either plain text or JSON. The new error text will be send as response body in
+  // either plain/text or application/json format for http, and in grpc-message header for grpc.
+  LocalReplyConfig local_reply_config = 38;
+}
+
+message LocalReplyConfig {
+  // Configuration of list of mappers which allows to filter and change local response.
+  // The client will iterate through mappers until first match.
+  repeated ResponseMapper mapper = 1;
+
+  // Specify how to form a new text body from access log :ref:`command operators <config_access_log_command_operators>`
+  // and in which format: either plain text or JSON. This is the catch-all format, it will be used
+  // if the matched mapper doesn't have its own format specified or there is not any mappers.
+  // If not specified, the text body will not be changed, the error detail passed in the
+  // sencLocalReply call will be used. The format will be plain text.
```

Comment:
typo "senc", also, and in any case, can you write this section from an end user perspective? I think you have the default behavior above.